src/cobra/util/solver.py

```diff
@@ -49,23 +55,26 @@
 has_primals = [NUMERIC, FEASIBLE, INFEASIBLE, SUBOPTIMAL, ITERATION_LIMIT, TIME_LIMIT]
 
 
-def linear_reaction_coefficients(model, reactions=None):
-    """Coefficient for the reactions in a linear objective.
+def linear_reaction_coefficients(
+    model: "Model", reactions: Optional[List["Reaction"]] = None
+) -> Union[Dict["Reaction", float], dict]:
```

Comment:
I don't understand right now why the union with `dict`?

Reply:
Because the function returns empty dictionary if there are no linear terms in the objective.

Reply:
An empty dictionary can still be of type `Dict["Reaction", float]` so you don't need the union.

Reply:
I see. Okay I'll make the change.